lib/matplotlib/axes/_axes.py

```diff
@@ -2292,6 +2292,14 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         xerr = kwargs.pop('xerr', None)
         yerr = kwargs.pop('yerr', None)
         error_kw = kwargs.pop('error_kw', {})
+        ezorder = error_kw.pop('zorder', None)
+        if ezorder is None:
+            ezorder = kwargs.pop('zorder', None)
+            if ezorder is not None:
+                # If using the bar zorder, increment slightly to make sure
+                # errorbars are drawn on top of bars
+                ezorder += 0.1
+        error_kw['zorder'] = ezorder
```

Comment:
we can use `error_kw.setdefault(ezorder)` which will only put the value in if the key does not exist (and is an atomic bytecode operation iirc).